design/ceph/admission-controller.md

```diff
@@ -0,0 +1,219 @@
+        --from-file=cert.pem=${PUBLIC_KEY_NAME}.pem
+```
+* Modifying the webhook config to inject CA bundle onto the ValidatingWebhookConfig
+* All the above resources will be created in rook-ceph namespace
+
+
+Using the above approach, the dev/admins will have the responsibility of rotating the certificates when they expire.
+
+Below is an excerpt of what a ValidatingWebhookConfig looks like 
+
+```
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: demo-webhook
+webhooks:
+  - name: webhook-server.webhook-demo.svc
+    clientConfig:
+      service:
+        name: webhook-server
+        namespace: rook-ceph
+        path: "/validate"
+      caBundle: ${CA_PEM_B64}
+    rules:
+      - operations: [ "CREATE" ]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+```
+
+We can make changes to the above values according to intercept based on whether a resource is being updated/deleted/created or change the type of resource or the request path which will be sent to the server.
+
+#### Production
+
+The onus of providing proper certificates and CA bundle for the server would be on the user. The following example would provide how it would work in an OCP cluster environment
+
+##### Openshift Container Platform
+
+![Prod_AC_Image](./rook ac-openshift.png)
+
+As shown in the above diagram, In the OCP cluster, The management of certificates would be completely done by the Openshift Service CA Operator. 
+
+The first step would be to generate the certificates and assign them to the proper components. On annotating any service with 
+
+`service.beta.openshift.io/serving-cert-secret-name:<secret-name>` 
+
+We have to deploy the the annotated service at the start to generate the signed certificates. This can done manually  through the command 
+
+`oc create -f ${SERVICE_NAME}.yaml`
+
+For verifying 
+
+`oc describe ${SERVICE_NAME} -n ${NAMESPACE}`
+
+**Output**
+```
+oc describe service <service-name>
+...
+Annotations:              service.beta.openshift.io/serving-cert-secret-name: <service-name>
+                          service.beta.openshift.io/serving-cert-signed-by: openshift-service-serving-signer@1556850837
+```
+The operator automatically generates a signed serving certificate and key pair into a secret in the same namespace as the service. 
+This secret will be mounted onto the server pod. Our server would be looking for tls certificates and automatically set them for TLS communication.
+For injecting a CA bundle onto the ValidatingWebhookConfig, we can annotate it with
+
+`service.beta.openshift.io/inject-cabundle=true`
+
+The operator on detecting this will set the ca bundle field in our config with a base64url-encoded CA signing bundle. These certificates will be automatically rotated and renewed by the service CA operator. This can also be done manually through the commands which can be found in the operator manual.
+
+### Starting the Webhook Server
```

Comment:
This is where the "Rook" heading should be, right?

Reply:
Fixed. Added "Starting the webhook server" part under "Setup in Rook"